include/hip/hcc_detail/device_library_decls.h

```diff
@@ -45,7 +45,8 @@ extern "C" __device__ float __ocml_trunc_f32(float);
 
 // Introduce local address space
 #define __local __attribute__((address_space(3)))
-__device__ inline static __local char* __to_local(unsigned x) { return (__local char*)x; }
+// TODO: Remove cast to uint64_t after clang bug is fixed. Should use uint32_t
+__device__ inline static __local char* __to_local(uint64_t x) { return (__local char*)x; }
```

Comment:
should the return type be `void*` instead of `char*`?

Reply:
Thanks, I will change that